internal/keystore/vault/client.go

```diff
@@ -141,12 +141,16 @@ type authFunc func(context.Context) (*vaultapi.Secret, error)
 //
 //	go client.RenewToken(ctx, login, ttl)
 func (c *client) RenewToken(ctx context.Context, authenticate authFunc, secret *vaultapi.Secret) {
-	ttl, _ := secret.TokenTTL()
+	s := secret
+	ttl, _ := s.TokenTTL()
 	if ttl == 0 {
 		return // Token has no TTL. Hence, we do not need to renew it. (long-lived)
 	}
 
-	const Retry = 3 // Retry token renewal N times before re-authenticating.
+	const (
+		Retry = 3                // Retry token renewal N times before re-authenticating.
+		Delay = 30 * time.Second // Wait a certain amount of time before using a new token to account for Vault replication lag
```

Comment:
should this be configurable? is 30sec enough?

Reply:
`Delay` is the time KES waits before it uses a new token received from one of the Vault nodes.
So one (actually N/2+1) Vault nodes consider this token as valid. However, due to replication lag some of the other (N/2-1) nodes may haven't received the token and cannot validate it. Hence, KES waits for 30s to allow other Vault nodes to catch up.

There is no "correct" value since Vault replication lag can be arbitrarily large. There could be a network partition between Vault nodes such that they "never" sync. However, 30s should be sufficient for all practical purposes. If you have higher replication lag, other issues arise most probably.